build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,239 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Context, Package } from "@fluidframework/build-tools";
+import async from "async";
+import chalk from "chalk";
+import execa from "execa";
+import clonedeep from "lodash.clonedeep";
+import { ResetMode } from "simple-git";
+
+import { packageOrReleaseGroupArg as baseArg, findPackageOrReleaseGroup } from "../../args";
+import { BaseCommand } from "../../base";
+import { Repository, getPreReleaseDependencies } from "../../lib";
+import { isReleaseGroup, ReleaseGroup, ReleasePackage } from "../../releaseGroups";
+
+type CheckFunction = (
+	context: Context,
+	rgOrPkg: ReleaseGroup | ReleasePackage,
+	print?: boolean,
+) => Promise<CheckResult>;
+
+interface CheckResult {
+	result: boolean;
+	message?: string;
+	fixCommand?: string;
+	fatal?: boolean;
+}
+
+const checkNoLocalChanges: CheckFunction = async (context: Context): Promise<CheckResult> => {
+	const status = await context.gitRepo.getStatus();
+	if (status !== "") {
+		return {
+			result: false,
+			message: "There are some local changes. The branch should be clean.",
+			fixCommand: "git stash",
+			fatal: true,
+		};
+	}
+
+	return { result: true };
+};
+
+const checkDepsInstalled: CheckFunction = async (
+	context: Context,
+	rgOrPkg: ReleaseGroup | ReleasePackage,
+	print?: boolean,
+): Promise<CheckResult> => {
+	const packagesToCheck = isReleaseGroup(rgOrPkg)
+		? context.packagesInReleaseGroup(rgOrPkg)
+		: // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		  [context.fullPackageMap.get(rgOrPkg)!];
+
+	async function checkDeps(pkg: Package): Promise<boolean> {
+		const depsAreInstalled = await pkg.checkInstall(print);
+		// console.warn(`deps for ${pkg.name}: ${depsAreInstalled}`);
+		return depsAreInstalled;
+		// return pkg.checkInstall(false);
+	}
+
+	// eslint-disable-next-line unicorn/no-array-method-this-argument, @typescript-eslint/no-misused-promises
+	const installed = await async.every(packagesToCheck, checkDeps);
+	if (!installed) {
+		return {
+			result: false,
+			message: "Some dependencies aren't installed. Try installing dependencies manually.",
+			fixCommand: "pnpm install",
+		};
+	}
+
+	return { result: true };
+};
+
+const checkHasRemote: CheckFunction = async (context: Context): Promise<CheckResult> => {
+	const remote = await context.gitRepo.getRemote(context.originRemotePartialUrl);
+	if (remote === undefined) {
+		return {
+			result: false,
+			message: "No remote found that points to the microsoft/FluidFramework repo.",
+		};
+	}
+	return { result: true };
+};
+
+const checkBranchUpToDate: CheckFunction = async (context: Context): Promise<CheckResult> => {
```

Comment:
none of these top level items have doc comments. Adding some can improve readability.

Reply:
Fixed.